rework css media-query (and lots of css fixes)

Carreau opened the pull request:
closes #7355
closes #6758

Also limit the number of query "jump", and use variables.

In particular there was a container width discontinuity at 768 pixel

where below 768 container width is 100% and "only" 720px on wider
screen, making a jump.

So now Css media query have 3 breakpoints:
- <540 px with  (aka -no name-)
  - colapsed menubar
  - vbox for prompt
  - 100% width container
- 540 -> 767 (aka -xs- )
  - expand menubar
  - hbox fro prompt
- 767 -> 991 (aka -sm-)
  - show save widget
  - fix width (768px)
- 991+ (aka -md-)
  - extra save widget info (date time)
  - fix with of 991 px
- 1200+ (aka -lg-)
  - widht of 1200px.
  
  of course we've set soem variable wrongly so this is still kinda-a-hack
  as we shoudl have pushed the xs-sm limit lower instead of pushing the
  nothing-xs limit up but I don't want to "fix it" now.

Carreau committed:
rework css media-query

fixes #7355

Also limit the number of query "jump", and use variables.

In particular there was a container width discontinuity at 768 pixel

where below 768 container width is 100% and "only" 720px on wider
screen, making a jump.

So now Css media query have 3 breakpoints:
    - <540 px with  (aka -no name-)
        - colapsed menubar
        - vbox for prompt
        - 100% width container
    - 540 -> 767 (aka -xs- )
        - expand menubar
        - hbox fro prompt
    - 767 -> 991 (aka -sm-)
        - show save widget
        - fix width (768px)
    - 991+ (aka -md-)
        - extra save widget info (date time)
        - fix with of 991 px.
    - 1200+ (aka -lg-)
        - widht of 1200px.

of course we've set soem variable wrongly so this is still kinda-a-hack
as we shoudl have pushed the xs-sm limit lower instead of pushing the
nothing-xs limit up but I don't want to "fix it" now.

minrk commented on IPython/html/static/base/less/page.less:
```diff
@@ -127,3 +132,9 @@ span#login_widget > .button,
 }
+
+@media (min-width: @screen-sm-min){
+    .container {
```
Why change the container width here?

Carreau replied:
Because otherwise I need to maintain a patch in `componenents/bootstrap/`.

I'm fine with dropping this if you don't like, but it's the part that make the notebook not jump from 768px width to 720 px width when you grow the window width from 768px to 769px.

minrk replied:
I wasn't objecting to the change, I was just curious what it addressed.

minrk replied:
This does change the maximum width of the notebook to 768px, when it used to be 1140px, so it's a dramatic change. Was that intentional? Or did you mean `min-width: @screen-sm-min` and `max-width: @screen-sm-max`?

Carreau replied:
> Or did you mean min-width: @screen-sm-min and max-width: @screen-sm-max?

Probably, let me re-check.

Carreau commented:
> Or did you mean min-width: @screen-sm-min and max-width: @screen-sm-max?
> Probably, let me re-check.

Was none of thoses. 

Fixed, rebased and PR description updated.
